.github/actions/setup/action.yml

```diff
@@ -1,26 +1,16 @@
-name: Setup
-
-inputs:
-  working-directory:
-    default: ./
-    required: false
+name: Install Yarn dependencies
+description: Installs the workspace's yarn dependencies and caches them
 
 runs:
   using: composite
   steps:
-    - name: Install node
-      uses: actions/setup-node@v3
-      with:
-        node-version: 18.17.1
```

Comment:
You really should be using setup-node to have specific versions of node instead of the outdated one that ships with actions.

Additionally, the setup-node action has the yarn caching directly built-in.